api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerCombatEvent.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * This file is part of packetevents - https://github.com/retrooper/packetevents
+ * Copyright (C) 2021 retrooper and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.github.retrooper.packetevents.wrapper.play.server;
+
+import com.github.retrooper.packetevents.event.PacketSendEvent;
+import com.github.retrooper.packetevents.protocol.packettype.PacketType;
+import com.github.retrooper.packetevents.protocol.player.Combat;
+import com.github.retrooper.packetevents.wrapper.PacketWrapper;
+
+import java.util.Optional;
+import java.util.OptionalInt;
+
+public class WrapperPlayServerCombatEvent extends PacketWrapper<WrapperPlayServerCombatEvent> {
+    private Combat combat;
+    private int duration;
+    private int entityId;
+    private int playerId;
+    private String deathMessage;
+
+    public WrapperPlayServerCombatEvent(PacketSendEvent event) {
+        super(event);
+    }
+
+    public WrapperPlayServerCombatEvent(Combat combat) {
+        super(PacketType.Play.Server.COMBAT_EVENT);
+        this.combat = combat;
+    }
+
+    public WrapperPlayServerCombatEvent(Combat combat, int duration, int entityId) {
+        super(PacketType.Play.Server.COMBAT_EVENT);
+        this.combat = combat;
+        this.duration = duration;
+        this.entityId = entityId;
+    }
+
+    public WrapperPlayServerCombatEvent(Combat combat, int playerId, int entityId, String deathMessage) {
+        super(PacketType.Play.Server.COMBAT_EVENT);
+        this.combat = combat;
+        this.playerId = playerId;
+        this.entityId = entityId;
+        this.deathMessage = deathMessage;
+    }
+
+    @Override
+    public void read() {
+        this.combat = Combat.VALUES[readVarInt()];
+        switch (combat) {
+            case END_COMBAT:
+                this.duration = readVarInt();
+                this.entityId = readInt();
+                break;
+            case ENTITY_DEAD:
+                this.playerId = readVarInt();
+                this.entityId = readInt();
+                this.deathMessage = readString();
```

Comment:
This changes cross version

Reply:
In which ones?

Reply:
Honestly, I can't find any changes cross versions